client/my-sites/invites/controller.js

```diff
@@ -17,7 +17,10 @@ export function acceptInvite( context ) {
 	context.layout.setState( { noSidebar: true } );
 
 	React.render(
-		React.createElement( InviteAccept, context.params ),
+		React.createElement(
+			InviteAccept,
+			context.params
```

Comment:
My guess is that if we moved to using `GET` parameters, the heavy lifting would be done here. In this method we would get the `GET` parameters above (See what I did there :stuck_out_tongue: :eyeglasses: ) and then we would do the `page.replace()` or similar.

The only other change I think we would need is updating the route in `my-sites/invites/index.js` to be `'/accept-invite/:site_id/:invitation_key`.